ctapipe_io_magic/__init__.py

```diff
@@ -889,6 +963,165 @@ def parse_metadata_info(self):
+                laser.Offset2 = laser_info_runh['MReportLaser.fOffset2']
+                laser.Offset3 = laser_info_runh['MReportLaser.fOffset3']
+                laser.Background1 = laser_info_runh['MReportLaser.fBackground1']
+                laser.Background2 = laser_info_runh['MReportLaser.fBackground2']
+                laser.BackgroundErr1 = laser_info_runh['MReportLaser.fBackgroundErr1']
+                laser.BackgroundErr2 = laser_info_runh['MReportLaser.fBackgroundErr2']
+                laser.RangeMax = laser_info_runh['MReportLaser.fRangeMax']
+                laser.RangeMax_Clouds = laser_info_runh['MReportLaser.fRangeMax_Clouds']
+                laser.ErrorCode = laser_info_runh['MReportLaser.fErrorCode']
+                laser.ScaleHeight_fit = laser_info_runh['MReportLaser.fScaleHeight_fit']
+                laser.Alpha_fit = laser_info_runh['MReportLaser.fChi2Alpha_fit']
+                laser.Chi2Alpha_fit = laser_info_runh['MReportLaser.fChi2Alpha_fit']
+                laser.Alpha_firstY = laser_info_runh['MReportLaser.fAlpha_firstY']
+                laser.Alpha_Junge = laser_info_runh['MReportLaser.fAlpha_Junge']
+                laser.PBLHeight = laser_info_runh['MReportLaser.fPBLHeight']
+                laser.Chi2Full_fit = laser_info_runh['MReportLaser.fChi2Full_fit']
+                laser.HWSwitchMaxOffset = laser_info_runh['MReportLaser.fHWSwitchMaxOffset']
+                laser.NCollapse = laser_info_runh['MReportLaser.fNCollapse']
+                laser.Shots = laser_info_runh['MReportLaser.fShots']
+                laser.T0Shift = laser_info_runh['MReportLaser.fT0Shift']
+                laser.Interval_0 = laser_info_runh['MReportLaser.fInterval_0']
+                laser.RCS_min_perfect = laser_info_runh['MReportLaser.fRCS_min_perfect']
+                laser.RCS_min_clouds = laser_info_runh['MReportLaser.fRCS_min_clouds']
+                laser.RCS_min_mol = laser_info_runh['MReportLaser.fRCS_min_mol']
+                laser.LIDAR_ratio = laser_info_runh['MReportLaser.fLIDAR_ratio']
+                laser.LIDAR_ratio_Cloud = laser_info_runh['MReportLaser.fLIDAR_ratio_Cloud']
+                laser.LIDAR_ratio_Junge = laser_info_runh['MReportLaser.fLIDAR_ratio_Junge']
+
+                mjd_values = laser_info_runh['MTimeLaser.fMjd']
+                millisec_values = laser_info_runh['MTimeLaser.fTime.fMilliSec']
+                for mjd_value, millisec_value in zip(mjd_values, millisec_values):
+                    millisec_seconds = millisec_value * 1e-3
+                    combined_mjd_value = mjd_value + millisec_seconds / 86400
+                    if (mjd_value, millisec_value) not in unique_reports:
+                        unique_reports[(mjd_value, millisec_value)] = combined_mjd_value
+            except KeyError as e:
+                print(f"Required key not found in the file {rootf}: {e}")
+                continue
+        for _ , value in unique_reports.items():
```

Comment:
I do not undeerstand the logic here.
Is ReportLaserContainer is supposed to store all the reports from the same file splitting them by variable and storing the arrays of those variables, right?

related to it:  here you do something strange, because you check for unique reports and then you loop and overwrite in each iteration the same value of MJD. So e.g. if you have 19 reports, but only 5 of them are really unique, and assuming that you store the arrays, all of them will have still all 19 entries (instead of 5 corresponding to unique MJDs), and the actual MJD will not store an array but only the MJD of the last report